### ISSUE TITLE
The ion score in fd mode is several times that of other Titans and should be limited to ion score

### ISSUE THREAD

**Issue by liveskai** [R2Northstar/NorthstarMods#653]
The ion score in fd mode is several times that of other Titans and should be limited to ion score

**Comment by ASpoonPlaysGames** [comment]
Can you be more specific? What do you mean by "ion score"?

**Comment by liveskai** [comment]
> 



> Can you be more specific? What do you mean by "ion score"?

Points earned for kills

**Comment by ASpoonPlaysGames** [comment]
> Points earned for kills

Like players killing enemy ions? Or player ions getting kills?

**Comment by Zanieon** [comment]
Combat Score probably, Splitter Rifle with Refraction Lens registers absurd damage amounts against enemy Titans because the score based on damage is not capped to their current health in the main gamemode_fd branch. I solved that issue in my branch and that will eventually get PR'd into main gamemode_fd.

**Comment by liveskai** [comment]
> > Points earned for kills
> 
> Like players killing enemy ions? Or player ions getting kills?

Players use ion to kill enemies

**Comment by liveskai** [comment]
> Combat Score probably, Splitter Rifle with Refraction Lens registers absurd damage amounts against enemy Titans because the score based on damage is not capped to their current health in the main gamemode_fd branch. I solved that issue in my branch and that will eventually get PR'd into main gamemode_fd.

laser core

**Comment by Dinorush** [comment]
@Zanieon Combat score in vanilla isn't capped by health either. I don't know what system it's using now or what yours uses, but vanilla combat score gain is

- Titans: 1 score per 100 damage (excludes shield damage)
- Reapers: 20 score
- Stalkers: 5 score
- Anything else: 1 score

**Comment by Zanieon** [comment]
@Dinorush Are you sure? Because i recall vanilla updates scoreboard only a bit before wave bonuses shows, and everything you killed or damaged in that wave comes condensed into either the Combat Score medal or the Support Score medal. FD has scoring consts in sh_consts.gnut between lines 2078~2104 and that is what i used for scoring, however Titans are treated different, they don't even give money per kill, they do per damage, and regardless of how much health Titans have it's always 50 money (no doomstate or shield).

Either way, if the score implementation vanilla does is different, then i'll have to change how i do it.

**Comment by Dinorush** [comment]
Yes. Those are values I got when I tested with one other person a while back. If I remember right, we would just drop into an FD private match duo and have one person kill/damage something specific and then let the second person do the rest of the wave to not let it influence their score.

Money is not tied to score so yeah, you'll get different numbers there. I'm honestly not sure what those values in sh_consts.gnut correspond to because they are certainly not the money values either.

**Comment by Zanieon** [comment]
I have changed the score implementation based on what Dinorush mentioned and no apparent change happened, Ion can still top score in the hands of a good player (especially if said player abuses the Splitter Glitch), but this absurd scoring might as well just happen to any other Titans, players using them just need to know the ways of doing absurd amounts of damage during the waves.

Case on point, problem is not Ion, just players playing well.
